### PR TITLE
[Module #4] solved tasks

### PR DESCRIPTION
**Pull request by pbochniak**
Every task solved in single commit as in module-3

### CODE REVIEW

File: module-4/task-3/refactor/refactor.go [hunk 1]
```diff
@@ -1,40 +1,58 @@
...
+	if e.err != nil {
+		return fmt.Sprintf("Error during file %q reading: %v", e.file, e.err.Error())
+	}
+	return fmt.Sprintf("Unknown error during file %q reading", e.file)
+}
+
+func readFile(path string) ([]byte, error) {
 	fd, err := os.OpenFile(path, os.O_RDONLY, os.ModePerm)
 	if err != nil {
-		panic(err)
```
Comment (yushni) [inline module-4/task-3/refactor/refactor.go:12]: Approach with Error types is also possible, but I think that this is the best solution in this code. Typically`return nil, fmt.Errorf("failed to open file: %w")` enough

File: module-4/task-2/panic/panic.go [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
 }
 
+func recoveryPanic() {
+	if err := recover(); err != nil {
+		fmt.Printf("recovery from panic - %q\n", err)
+	}
+}
+
 func catchPanic() {
+	defer recoveryPanic()
```
Comment (yushni) [inline module-4/task-2/panic/panic.go:16]: it is not necessary to create new function, typically we do 
```
defer func() {
   recovery code 
} () 
```

File: module-4/task-3/refactor/refactor.go (hunk 1 above)
Comment (pbochniak) [reply module-4/task-3/refactor/refactor.go:12]: I have just use better practic according to Alieksei lecture

File: module-4/task-2/panic/panic.go (hunk 2 above)
Comment (pbochniak) [reply module-4/task-2/panic/panic.go:16]: OK, I have considered that option, but finally I chose to wriet a function.

File: module-4/task-3/refactor/refactor.go (hunk 1 above)
Comment (yushni) [reply module-4/task-3/refactor/refactor.go:12]: This solution also possible, and it is prefered when you rely on the type or have extra data on the error. But usually, it is overhead to write an extra type for every single error.

**Review by yushni — APPROVED** [review]

**Pull request merged**